### PR TITLE
Added New Sample - Document AI Warehouse Processing

### PR DESCRIPTION
**Pull request by HSbedi87**
Added New Sample - Document AI Warehouse Processing

### CODE REVIEW

File: document_ai_warehouse_processing_python/document_warehouse_utils.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+        request.name = f"{parent}/documentSchemas/{schema_id}"
+
+        # Make the request
+        response = client.delete_document_schema(request=request)
+
+        # Handle the response
+        return response
+
+    def update_document_schema(self, schema_id: str, schema: str):
+        schema_json = json.loads(schema)
```
Comment (holtskinner) [inline document_ai_warehouse_processing_python/document_warehouse_utils.py:350]: Same here with json conversion.

There are parameters you can add to control if it ignores unknown fields if needed

`ignore_unknown_fields: bool = False`

**Review by holtskinner — CHANGES_REQUESTED** [review]
LGTM, just a few minor things to make the code easier to follow

**Comment by holtskinner** [comment]
Also, fix linter errors

File: document_ai_warehouse_processing_python/document_warehouse_utils.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+            document.raw_document_file_type = document.raw_document_file_type.RAW_DOCUMENT_FILE_TYPE_DOCX
+        elif mime_type.lower() == 'text/plain':
+            document.raw_document_file_type = document.raw_document_file_type.RAW_DOCUMENT_FILE_TYPE_TEXT
+        elif mime_type.lower() == 'application/vnd.openxmlformats-officedocument.presentationml.presentation':
+            document.raw_document_file_type = document.raw_document_file_type.RAW_DOCUMENT_FILE_TYPE_PPTX
+        elif mime_type.lower() == 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet':
+            document.raw_document_file_type = document.raw_document_file_type.RAW_DOCUMENT_FILE_TYPE_XLSX
+        else:
+            document.raw_document_file_type = document.raw_document_file_type.RAW_DOCUMENT_FILE_TYPE_UNSPECIFIED
+
```
[the remark this answers is not in the document]
Comment (HSbedi87) [reply document_ai_warehouse_processing_python/document_warehouse_utils.py:181]: Updated by using dict

File: document_ai_warehouse_processing_python/document_warehouse_utils.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+    def apped_docai_entities_to_doc_properties(self,cloud_ai_document: docai.Document, document: contentwarehouse_v1.Document, docai_property_name: str):
```
[the remark this answers is not in the document]
Comment (HSbedi87) [reply document_ai_warehouse_processing_python/document_warehouse_utils.py:182]: Fixed it

File: document_ai_warehouse_processing_python/document_warehouse_utils.py (hunk 1 above)
Comment (HSbedi87) [reply document_ai_warehouse_processing_python/document_warehouse_utils.py:350]: Changed to contentwarehouse_v1.DocumentSchema.from_json() for json conversion

File: document_ai_warehouse_processing_python/document_warehouse_utils.py [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+
+    def delete_document_schema(self,schema_id: str):
+
+        client = self.get_document_schema_service_client()
+        parent = client.common_location_path(self.project_number, self.api_location)
+
+        # Initialize request argument(s)
+        request = contentwarehouse_v1.DeleteDocumentSchemaRequest()
+
+        request.name = f"{parent}/documentSchemas/{schema_id}"
```
[the remark this answers is not in the document]
Comment (HSbedi87) [reply document_ai_warehouse_processing_python/document_warehouse_utils.py:305]: I tried to search but could not find it.

**Review by HSbedi87 — COMMENTED** [review]
I have made changes to the code.

File: document_ai_warehouse_processing_python/.gitignore [hunk 5]
```diff
@@ -126,4 +126,6 @@ venv.bak/
 dmypy.json
 
 # Pyre type checker
-.pyre/
\ No newline at end of file
+.pyre/
+
+main.py
```
Comment (holtskinner) [inline document_ai_warehouse_processing_python/.gitignore:131]: is there a reason to ignore `main.py`?

File: document_ai_warehouse_processing_python/document_warehouse_utils.py [hunk 6]
```diff
@@ -355,14 +322,7 @@ def update_document_schema(self, schema_id: str, schema: str):
...
         # define schema
-        request.document_schema = contentwarehouse_v1.DocumentSchema()
-        request.document_schema.display_name = schema_json["display_name"]
-        request.document_schema.document_is_folder = schema_json["document_is_folder"]
-        request.document_schema.description = schema_json["description"]
-
-        for property_definition in schema_json["property_definitions"]:
-            request.document_schema.property_definitions.append(
-                self.create_document_schema_property_definition(property_definition))
+        request.document_schema = contentwarehouse_v1.DocumentSchema.from_json(schema_json)
```
Comment (holtskinner) [inline document_ai_warehouse_processing_python/document_warehouse_utils.py:342]: Glad this worked! Greatly simplified this function.

(This type of conversion isn't obvious in the client libraries)

File: document_ai_warehouse_processing_python/requirements.txt [hunk 7]
```diff
@@ -3,5 +3,4 @@ google-cloud-core==2.3.2
 google-cloud-documentai==2.4.1
```
Comment (holtskinner) [inline document_ai_warehouse_processing_python/requirements.txt:3]: Minor: The latest version is `2.6.0` (Note: The library versions will be auto-updated in this repo)

**Review by holtskinner — APPROVED** [review]

**Comment by holtskinner** [comment]
Just a few more linter errors

Looks like it uses

- markdownlint
  - https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint
- mypy
- flake8

These can be installed or enabled in vscode.
(I'm going to ask DEE if there's a config file that can be used that sets the same requirements as this linter)

**Review by holtskinner — CHANGES_REQUESTED** [review]
Resolve linter errors and remove `main.py` from `.gitignore`

**Comment by HSbedi87** [comment]
Fixed linter issues and removed main.py form .gitignore

**Review by holtskinner — APPROVED** [review]

**Pull request merged**